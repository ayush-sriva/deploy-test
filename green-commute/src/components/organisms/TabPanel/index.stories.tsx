import { ComponentStory, ComponentMeta } from '@storybook/react';
import TabPanel from '.';

export default {
  title: 'components/organisms/Tab Panel',
  component: TabPanel,
} as ComponentMeta<typeof TabPanel>;

const Template: ComponentStory<typeof TabPanel> = (args) => <TabPanel {...args} />;

export const TabPanelDemo = Template.bind({});

TabPanelDemo.args = {};
