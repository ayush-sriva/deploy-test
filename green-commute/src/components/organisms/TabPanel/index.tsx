/**
 * @author Abhishek Bhemisetty <abhishek.bhemisetty@zemosolabs.com>
 */

import Grid from '@mui/material/Grid';
import ProgressItem from '../../molecules/ProgressItem';
import DoneIcon from '@mui/icons-material/Done';
import { styled } from '@mui/material';

interface TabProps {
  pageId?: number;
}

interface ProgressProps {
  value: string | React.ReactNode;
  label: string;
  status: string;
}

const TabItem = styled(Grid)({
  marginRight: '15px',
});

const TabPanel: React.FC<TabProps> = ({ pageId }) => {
  const Tabdata: Array<ProgressProps> = [
    {
      value: '1',
      label: 'Your Location',
      status: 'Current',
    },
    {
      value: '2',
      label: 'Job Location',
      status: 'Future',
    },
    {
      value: '3',
      label: 'Your Skills',
      status: 'Future',
    },
  ];

  if (pageId === 1) {
    Tabdata[0].value = <DoneIcon />;
    Tabdata[1].status = 'Current';
  } else if (pageId === 2) {
    Tabdata[0].value = <DoneIcon />;
    Tabdata[1].status = 'Current';
    Tabdata[1].value = <DoneIcon />;
    Tabdata[2].status = 'Current';
  } else {
    Tabdata[1].status = 'Future';
    Tabdata[1].value = '2';
    Tabdata[2].status = 'Future';
    Tabdata[2].value = '3';
  }

  const Tabs = () => {
    return Tabdata.map((tab: ProgressProps, id: number) => {
      return (
        <TabItem item key={id}>
          <ProgressItem value={tab.value} label={tab.label} status={tab.status}></ProgressItem>
        </TabItem>
      );
    });
  };

  return (
    <Grid container spacing={2} direction='row'>
      {Tabs()}
    </Grid>
  );
};

export default TabPanel;
