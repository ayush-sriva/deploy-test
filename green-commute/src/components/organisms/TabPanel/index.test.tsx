import ReactTestRenderer from 'react-test-renderer';
import TabPanel from '.';

describe('Tab Panel Test', () => {
  it('renders the tab panel', () => {
    const tree = ReactTestRenderer.create(<TabPanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
