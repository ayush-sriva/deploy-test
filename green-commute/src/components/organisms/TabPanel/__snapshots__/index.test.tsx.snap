// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Panel Test renders the tab panel 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item css-1p1lgg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      data-testid="item-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        >
          1
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignJustify css-669lo9-MuiTypography-root"
        >
          Your Location
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-1p1lgg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      data-testid="item-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        >
          2
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignJustify css-tv3ey-MuiTypography-root"
        >
          Job Location
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item css-1p1lgg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      data-testid="item-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        >
          3
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignJustify css-tv3ey-MuiTypography-root"
        >
          Your Skills
        </h6>
      </div>
    </div>
  </div>
</div>
`;
